lib/account.rb

```diff
@@ -3,53 +3,79 @@
 # Dependency: Ruby money -- gem install money
 require 'csv'
 require 'money'
+require 'colorize'
 I18n.enforce_available_locales = false
 
 module Bank
 
   class Account
-    attr_reader :balance, :id, :owner, :open_date
-    # Instantiation of object has optional parameters of balance and owner
-    def initialize(id, balance, open_date, owner = nil)
+    attr_reader :balance, :id, :owner, :open_date, :type
+    FEE = 0
+    MIN_BALANCE = 0
+    # Instantiation of object has optional parameters of open_date and owner
+    def initialize(id, balance, open_date = "today", owner = nil)
       @owner = owner
-      # Creates an ID of random numbers
-      @id = id
-      # Raises an error with a rescue for a negative initial balance
-      if balance < 0
-        begin
-          raise ArgumentError.new("You may not create an account with a negative balance.")
-        rescue
-          puts "Setting balance to a default value of 0."
-          @balance = 0
-        end
-      else
-        @balance = balance
+      @id = id.to_i
+      @open_date = open_date
+      @type = "Standard"
+      @balance = balance.to_i
+      # Raises an error if the balance is below the minimum balance
+      if balance.to_i < self.class::MIN_BALANCE
+          raise ArgumentError.new("You may not create an account below the minimum balance.")
       end
     end
 
-    def withdraw(amount)
-      if @balance - amount < 0
-        puts "You cannot withdraw more than your account balance."
+    # The withdraw method withdraws from the account
+    # check_min parameter - BOOLEAN
+    #   whether to check if the balance minus amount & fee is less than the minimum balance
+    # do_penalty parameter - BOOLEAN
+    #   whether to subtract the fee from the balance
+    def withdraw(amount, check_min = true, do_penalty = true)
+      if amount < 0
+        puts "You cannot withdraw a negative amount of money."
       else
-        puts "Starting balance: " + Money.new(@balance, "USD").format
-        puts "Amount withdrawn: " + Money.new(amount, "USD").format
-        @balance -= amount
-        puts "Updated balance: " + Money.new(@balance, "USD").format
+        puts "------#{@type.upcase} WITHDRAWAL------".colorize(:blue)
+        # If the withdrawal will put balance below 0, don't do it an output an error
+        if @balance - amount < 0
+          puts "You do not have sufficient funds to withdraw that amount."
+          # This check is done with standard, checking, and savings accounts
+        elsif @balance - (amount + self.class::FEE) < self.class::MIN_BALANCE && check_min
+          puts "You cannot go below the minimum account balance of " + Money.new(self.class::MIN_BALANCE, "USD").format
+          # This section does the actual withdrawal
+        else
+          puts "Starting balance: " + Money.new(@balance, "USD").format
```

Comment:
It looks like there is a bunch of code where you call this same Money logic so it may be a good place to create a method to take care of the formatting for you